resolver/resolver.go

```diff
@@ -3421,14 +3451,13 @@ func (r *Resolver) getEnvMessage(msg *Message) *Message {
 			if svc.Rule.Env == nil {
 				continue
 			}
-			// It will be created using the Env definition of the first matched service.
-			// TODO: Since it might match multiple services,
-			// it's necessary to verify whether there are any fields in the Env that are not common among those services.
-			// Currently, it targets services that exist in the same package as the message.
-			return envToMessage(svc.File, svc.Rule.Env)
+			envSvcs = append(envSvcs, svc)
 		}
 	}
-	return nil
+	if len(envSvcs) != 1 {
```

Comment:
nit, The following three lines may be a little cryptic for others, so it may be better to leave some comments here